### PR TITLE
[SCB-985] Add spring-boot2-starter-discovery

### PR DESCRIPTION
**Pull request by yangbor**
The code is copied from springboot-starter-discovery and test code
is removed. Adjusted the springboot and spring cloud versions for
compatibility issues.

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: spring-boot2-starter-parent/spring-boot2-starter-discovery/src/main/java/org/apache/servicecomb/springboot2/starter/discovery/CseDiscoveryClient.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    context.setInputParameters(serviceId);
+    DiscoveryTree discoveryTree = discoveryTrees.computeIfAbsent(serviceId, key -> {
+      return new DiscoveryTree();
+    });
+    VersionedCache serversVersionedCache = discoveryTree.discovery(context,
+        RegistryUtils.getAppId(),
+        serviceId,
+        DefinitionConst.VERSION_RULE_ALL);
+    Map<String, MicroserviceInstance> servers = serversVersionedCache.data();
+    List<ServiceInstance> instances = new ArrayList<>(servers.size());
```
Comment (wujimin) [inline spring-boot2-starter-parent/spring-boot2-starter-discovery/src/main/java/org/apache/servicecomb/springboot2/starter/discovery/CseDiscoveryClient.java]: this convert result should be cached
please create a issue to trace the optimize

File: java-chassis-dependencies-springboot2/pom.xml [hunk 2]
```diff
@@ -37,10 +37,11 @@
     <argLine>-Dfile.encoding=UTF-8</argLine>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
 
-    <spring.version>5.0.4.RELEASE</spring.version>
-    <spring.boot.version>2.0.0.RELEASE</spring.boot.version>
-    <spring.data.version>2.0.5.RELEASE</spring.data.version>
+    <spring.version>5.0.6.RELEASE</spring.version>
```
Comment (liubao68) [inline java-chassis-dependencies-springboot2/pom.xml]: spring & spring boot use version 5.0.7.RELEASE & 2.0.4.RELEASE

File: java-chassis-dependencies-springboot2/pom.xml [hunk 3]
```diff
@@ -147,6 +148,42 @@
         <artifactId>spring-session-data-redis</artifactId>
         <version>${spring.session.version}</version>
       </dependency>
+      <dependency>
+        <groupId>org.springframework.cloud</groupId>
+        <artifactId>spring-cloud-dependencies</artifactId>
```
Comment (liubao68) [inline java-chassis-dependencies-springboot2/pom.xml]: for spring boot & spring cloud, use their management like this 

            <dependency>
                <groupId>org.springframework.boot</groupId>
                <artifactId>spring-boot-dependencies</artifactId>
                <version>${spring.boot.version}</version>
                <type>pom</type>
                <scope>import</scope>
            </dependency>

File: java-chassis-dependencies-springboot2/pom.xml [hunk 4]
```diff
@@ -147,6 +148,42 @@
...
+        <version>${spring-cloud.version}</version>
+      </dependency>
+
+      <dependency>
+        <groupId>org.springframework.cloud</groupId>
+        <artifactId>spring-cloud-commons</artifactId>
+        <version>2.0.0.RELEASE</version>
```
Comment (liubao68) [inline java-chassis-dependencies-springboot2/pom.xml]: These dependencies can be deleted if you trying the way provided above.

File: java-chassis-dependencies-springboot2/pom.xml [hunk 5]
```diff
@@ -37,10 +37,11 @@
...
-    <spring.boot.version>2.0.0.RELEASE</spring.boot.version>
-    <spring.data.version>2.0.5.RELEASE</spring.data.version>
+    <spring.version>5.0.6.RELEASE</spring.version>
+    <spring.boot.version>2.0.3.RELEASE</spring.boot.version>
+    <spring.data.version>2.0.3.RELEASE</spring.data.version>
     <spring.session.version>2.0.3.RELEASE</spring.session.version>
+    <spring-cloud.version>Finchley.RELEASE</spring-cloud.version>
   </properties>
 
   <dependencyManagement>
```
Comment (liubao68) [inline java-chassis-dependencies-springboot2/pom.xml]: <dependencies>
       <dependency>	       <dependency>
         <groupId>org.springframework.boot</groupId>	         <groupId>org.springframework.boot</groupId>
         <artifactId>spring-boot-starter</artifactId>	         <artifactId>spring-boot-starter</artifactId>
         <version>${spring.boot.version}</version>	         <version>${spring.boot.version}</version>
         <type>pom</type>	         <type>pom</type>
         <scope>import</scope>	         <scope>import</scope>
       </dependency>	       </dependency>
       <dependency>	       <dependency>
         <groupId>org.springframework.boot</groupId>	         <groupId>org.springframework.boot</groupId>
         <artifactId>spring-boot-starter-web</artifactId>	         <artifactId>spring-boot-starter-web</artifactId>
         <version>${spring.boot.version}</version>	         <version>${spring.boot.version}</version>
         <type>pom</type>	         <type>pom</type>
         <scope>import</scope>	         <scope>import</scope>
       </dependency>	       </dependency>

Can be deleted if using spring boot dependencies.

Comment (yangbor) [reply java-chassis-dependencies-springboot2/pom.xml]: All redundant dependencies were cleaned up.

**Pull request merged**